test/bh/mix-test.js

```diff
@@ -0,0 +1,132 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('mix()(×)', function() {
+  it('should return mix', function() {
+    compile(function() {
+      block('button').def()(function() {
+        return JSON.stringify(this.ctx.mix);
+      });
+    })
+      .apply({ block: 'button', mix: { block: 'mix' } })
+      .should.equal('{"block":"mix"}');
+  });
+
+  it('should set single mix', function() {
+    compile(function() {
+      block('button').mix()({ block: 'mix' })
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set single mix as function', function() {
+    compile(function() {
+      block('button').mix()(function() { return { block: 'mix' }; });
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set array mix', function() {
+    compile(function() {
+      block('button').mix()([ { block: 'mix' } ]);
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  it('should set array mix as function', function() {
+    compile(function() {
+      block('button').mix()(function() { return [ { block: 'mix' } ]; });
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  xit('should extend user single mix', function() {
+    compile(function() {
+      block('button').mix()({ block: 'mix2' });
+    })
+      .apply({ block: 'button', mix: { block: 'mix1' } })
+      .should.equal('<div class="button mix1 mix2"></div>');
+  });
+
+  xit('should extend user array mix', function() {
+    compile(function() {
+      block('button')(
+        mix()([ { block: 'mix' } ])
+      );
+    })
+      .apply({ block: 'button', mix: [ { block: 'user-mix' } ] })
+      .should.equal('<div class="button user-mix mix"></div>');
+  });
+
+  xit('should extend later declarations', function() {
+    compile(function() {
+      block('button')(
+        mix()({ block: 'mix2' })
+      );
+      block('button')(
+        mix()({ block: 'mix1' })
+      );
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix1 mix2"></div>');
+  });
+
+  xit('should override later declarations with force flag', function() {
+    compile(function() {
+      block('button').force()(
+        mix()({ block: 'mix2' })
+      );
+      block('button')(
+        mix()({ block: 'mix1' })
+      );
+    })
+      .apply({ block: 'button' })
+      .should.equal('<div class="button mix2"></div>');
+  });
+
+  xit('should override user declarations with force flag', function() {
+    compile(function() {
+      block('button').force().mix()({ block: 'mix' });
+    })
+      .apply({ block: 'button', mix: { block: 'user-mix' } })
+      .should.equal('<div class="button mix"></div>');
+  });
+
+  xit('should inherit block name', function() {
```

Comment:
I think [it’s work well](http://bem.github.io/bem-xjst/?bemhtml=block(%27button%27).elem(%27control%27)(%0A%20%20%20%20mix()(%5B%0A%20%20%20%20%20%20%20%20%7B%20mods%3A%20%7B%20disabled%3A%20true%20%7D%20%7D%2C%0A%20%20%20%20%20%20%20%20%7B%20elem%3A%20%27input%27%2C%20mods%3A%20%7B%20active%3A%20true%20%7D%20%7D%2C%0A%20%20%20%20%20%20%20%20%7B%20block%3A%20%27clearfix%27%20%7D%0A%20%20%20%20%5D)%0A)%3B%0A%0A%0A&bemjson=(%7B%20block%3A%20%27button%27%2C%20elem%3A%20%27control%27%20%7D)%3B). But expected result in test is wrong.